core/src/main/scala/slamdata/AutoBump.scala

```diff
@@ -0,0 +1,369 @@
+      log.warn("you should check for a stuck trickle PR on that repository")
+    }
+  }
+  case object PushError extends Warnings {
+    def warn(log: Logger): IO[Unit] = IO {
+      log.warn("git-push exited with error")
+      log.warn("this usually means some other repository updated the pull request before this one")
+    }
+  }
+  case class NotOldest(oldest: PullRequestDraft, draft: PullRequestDraft) extends Warnings {
+    def warn(log: Logger): IO[Unit] = IO {
+      log.warn(s"pull request ${draft.number} is newer than existing pull request ${oldest.number}")
+      log.warn("this usually means two or more repositories finished build at the same time,")
+      log.warn("and some other repository beat this one to pull request creation.")
+    }
+  }
+
+  sealed abstract class ChangeLabel(val label: String)
+  object ChangeLabel {
+    val values: List[ChangeLabel] = List(RevisionChange, FeatureChange, BreakingChange)
+    val order: Map[ChangeLabel, Int] = values.zipWithIndex.toMap
+    implicit val ordering: Ordering[ChangeLabel] = Ordering.by(order)
+    val fromString: Map[String, ChangeLabel] = values.map(change => change.label -> change).toMap
+    val labelPattern: Regex = values.mkString("|").r
+
+    def apply(label: String): Option[ChangeLabel] = {
+      fromString.get(label)
+    }
+
+    def unapply(arg: String): Option[ChangeLabel] = labelPattern.findFirstIn(arg).flatMap(apply)
+  }
+  case object RevisionChange extends ChangeLabel("version: revision")
+  case object FeatureChange extends ChangeLabel("version: feature")
+  case object BreakingChange extends ChangeLabel("version: breaking")
+
+  implicit class Unpack[F[_]: Monad, A](response: F[GHResponse[A]]) {
+    def unpack: F[Either[GHException, A]] = EitherT(response).map(_.result).value
+  }
+
+  val AutoBumpLabel = ":robot:"
+  val PullRequestFilters: List[PRFilter] = List(PRFilterOpen, PRFilterSortCreated, PRFilterOrderAsc, PRFilterBase("master"))
+  val LinkRelation: Regex = """<(.*?)>; rel="(\w+)"""".r
+  val LabelPattern: Regex = "version: \\w+".r
+
+  def autoBumpCommitTitle(author: String) = s"Applied dependency updates by $author"
+
+  def autoPage[F[_]: Sync, T](first: Pagination)
+                             (call: Pagination => F[Either[GHException, GHResult[List[T]]]]): Stream[F, T] = {
+    val chunker = call.andThen(_.rethrow.map(res => nextPage(getRelations(res.headers)).map(Chunk.seq(res.result) -> _)))
+    Stream.unfoldChunkEval(first)(chunker)
+  }
+
+  def nextPage(relations: Map[String, (Int, Int)]): Option[Pagination] = {
+    relations.get("next").map((Pagination.apply _).tupled)
+  }
+
+  /** Decodes github's "Link" header into a map */
+  def getRelations(headers: Map[String, String]): Map[String, (Int, Int)] = {
+    val relations = for {
+      header <- headers.map { case (k, v) => k.toLowerCase -> v }.collect { case ("link", header) => header }
```

Comment:
More random style nits… *Either* use `.` and `(`, or use <kbd>space</kbd> and `{`, but not a mixture of the two. So the above should be the following if you want to keep it one line:

```suggestion
      header <- headers map { case (k, v) => k.toLowerCase -> v } collect { case ("link", header) => header }
```

Or you can break it up into two lines for readability, but in either case you're going to end up not using `.`.

The above actually nicely illustrates *why* this rule is something you should adhere to: depending on syntactic context, `}.` doesn't do what you think it does. In this case, it *is* doing the "right" thing, but in other contexts it won't, and the rules around it are very subtle. Not quite as bad as in Ruby, but still bad. Just avoid it.

Reply:
I'm well familiar with parenthesis vs braces in Scala. I thought you didn't like operator notation on non-symbolic methods.